src/qengine/gates.cpp

```diff
@@ -24,6 +24,14 @@ void QEngineCPU::SetBit(bitLenInt qubit1, bool value)
     }
 }
 
+/// Swap values of two bits in register
+void QEngineCPU::Swap(bitLenInt qubit1, bitLenInt qubit2)
```

Comment:
Qinterface?

Reply:
We covered this in the last PR. No, it would be circular, with a bitwise parallel gate implementation pointing to this and vice versa. I'll make sure the default bitwise method is in the interface, though.